From 95f8b55c62eb2cac8f331913aced9f8e5d849f70 Mon Sep 17 00:00:00 2001
From: slwyts <slwyts@foxmail.com>
Date: Tue, 20 Oct 2026 14:10:00 +0000
Subject: [PATCH] fix(gunyah): use Linux 7.3 reverse mapping iterator

Keep the mapping lock and reject folios covered by any VMA before lending. The iterator takes an address_space and an inclusive last page index.
---
 drivers/virt/gunyah/vm_mgr.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/virt/gunyah/vm_mgr.h b/drivers/virt/gunyah/vm_mgr.h
index 84f0e60b0cc..3021681c034 100644
--- a/drivers/virt/gunyah/vm_mgr.h
+++ b/drivers/virt/gunyah/vm_mgr.h
@@ -173,8 +173,8 @@ static bool folio_mmapped(struct folio *folio)
 	bool ret = false;
 
 	i_mmap_lock_read(mapping);
-	vma_interval_tree_foreach(vma, &mapping->i_mmap, folio->index,
-				  folio->index + folio_nr_pages(folio)) {
+	mapping_rmap_tree_foreach(vma, mapping, folio->index,
+				 folio->index + folio_nr_pages(folio) - 1) {
 		ret = true;
 		break;
 	}
-- 
2.56.0

